Source/FFmpegMediaSource.cpp

```diff
@@ -741,11 +741,27 @@ namespace winrt::FFmpegInteropX::implementation
             }
 
             //inject custom properties
-            if (config->AutoCorrectAnsiSubtitles() && config->IsExternalSubtitleParser && streamByteOrderMark != ByteOrderMark::UTF8)
+            if (config->AutoCorrectAnsiSubtitles() && config->IsExternalSubtitleParser &&
+                (streamEncoding != TextEncodingDetect::UTF8_BOM && streamEncoding != TextEncodingDetect::UTF8_NOBOM && streamEncoding != TextEncodingDetect::None))
             {
                 hstring key = config->AnsiSubtitleEncoding().Name();
                 std::string keyA = StringUtils::PlatformStringToUtf8String(key);
-                const char* keyChar = keyA.c_str();
+                const char* keyChar;
```

Comment:
It should be documented that `AnsiSubtitleEncoding` is only in effective in case of auto-detection and that when somebody wants to set the encoding explicitly, they need to disable auto-detection and set `sub_charenc` directly as ffmpeg parameter.

Reply:
Ack

Reply:
Well, sub_charenc must be set on the AVCodecContext of the stream decoder. There is currently no way to set decoder specific options in our lib.

But we could achieve this by slightly changing the meaning of the option. As I see it, there is really no good reason to disable auto detect. The only plausible reason would be if you want to set the encoding explicitly. So I think if the auto detection is false, we should just set the encoding from the config. It makes much more sense like that. Disabling auto detect without setting an explicit encoding just does not make sense. And in the end, it would force reading as UTF8. If someone really wants that, they could just disable auto detect and set the encoding to UTF8. So we would not prevent anyone from assuming UTF8 without any checking.

Reply:
Or we remove the boolean and have two properties of type CharacterEncoding: SubtitleEncoding and AnsiSubtitleEncoding. SubtitleEncoding defaults to null, which means auto detection. AnsiSubtitleEncoding is used if ANSI is detected. If SubtitleEncoding is set to a value, it is always used (no auto detection).

Maybe that is even better. Properly describing the behavior of the boolean as in my last post is pretty complicated (I just tried to write a comment for it). This one would be much easier.

Reply:
An ideal approach IMO would be as follows:

- An option AsciiCodePage` (default: "from system locale")
- An option TextDecoding with values: (default: Automatic)
  - Automatic
  - UTF-8
  - ASCII
 
### UTF-8

- No checks are performed
- sub_charenc: `utf8`
- sub_charenc_mode: `FF_SUB_CHARENC_MODE_IGNORE`


### ASCII

- No checks are performed
- sub_charenc:  [AsciiCodePage option]
- sub_charenc_mode: `FF_SUB_CHARENC_MODE_PRE_DECODER`

### Automatic

- Check for BOM
  If present, do like UTF-8 above

- Check content for UTF-8
  If positive, do like UTF-8 above

- Otherwise
  Do like ASCII above

Reply:
That would only allow setting two fixed encodings, plus auto. If someone wants to force a different encoding, they are screwed. With my other solution, any encoding can be forced (including UTF-8 and ASCII). If UTF-8 is forced, we can of course set the charenc_mode as proposed.

Reply:
Another idea would be to merge the auto detection in the existing CharacterEncoding class. This would mean that for each code page we would have a "Force <code page>" and "Auto detect <code page>" property (and double the number of possible CharacterEncoding instances). 

And something slightly unrelated: do we want to support per stream encoding settings? 
Technically for external subs we could add CharacterEncoding overload. And we should probably expose the CharacterEncoding property in the stream. This could be a different PR.

Reply:
I pushed an update showing the config options I proposed. I think this should cover almost any scenario.

Reply:
Looks good to me.

Reply:
> That would only allow setting two fixed encodings, plus auto. If someone wants to force a different encoding, they are screwed. Sorry, for being unclear. This involves TWO settings:

- An option AsciiCodePage` (default: "from system locale")
- An option TextDecoding with values: (default: Automatic)
  - Automatic
  - UTF-8
  - ASCII

Reply:
Actually it doesn't matter that much as long as Auto is the default and sys locale the default fallback for ASCII.
(ANSI are **only those 128 char which are never changing**),.

Reply:
You mean ASCII. ASCII only has these fixed 128 characters, as opposed to exended ASCII, which opens up the other 128 characters. The ANSI has standardized a lot of extended ASCII encodings. Although most of the Windows code pages are not ANSI standards but proprietary, the term ANSI Code Page or ANSI encoding has become quite common for describing extended ASCII character encodings, so we should stick with that. Also the encoding detection lib returns "ANSI" if it sees use of the extended ASCII range.

Reply:
Sorry, I was mistaken on that point, even though I had quickly verified before posting (at the time weeks ago). Seems I had come to a place which had got it wrong.

This table sums it up pretty well:

![image](https://github.com/ffmpeginteropx/FFmpegInteropX/assets/4985349/7f903fa1-207e-49bd-9934-ae504fe28f24)

I am very sorry for the confusion. Unless it's accompanied by 'think', 'believe' or something, I usually verify things before I'm writing (the world is full enough of misinformation 😉), but this time I failed, as I thought it would be other way round and the first confirmation I found seemed to suffice.

Reply:
Added all required documentation